animation_nodes/nodes/geometry/c_utils.pyx

```diff
@@ -0,0 +1,269 @@
+            scaleVec3(&intersectionVector, &direction, factor)
+            addVec3(&_intersection, lineStart.get(i), &intersectionVector)
+            intersection.data[i] = _intersection
+            parameter.data[i] = factor
+            valid.data[i] = True
+        else:
+            intersection.data[i] = Vector3(0,0,0)
+            parameter.data[i] = 0
+            valid.data[i] = False
+    return intersection, parameter, valid
+
+def IntersectLineLine(Py_ssize_t amount,
+                    VirtualVector3DList firstLineStart,
+                    VirtualVector3DList firstLineEnd,
+                    VirtualVector3DList secondLineStart,
+                    VirtualVector3DList secondLineEnd):
+    cdef Vector3DList nearestPoint1 = Vector3DList(length = amount)
+    cdef Vector3DList nearestPoint2 = Vector3DList(length = amount)
+    cdef DoubleList firstParameter = DoubleList(length = amount)
+    cdef DoubleList secondParameter = DoubleList(length = amount)
+    cdef BooleanList valid = BooleanList(length = amount)
+    cdef Vector3 direction1, direction2, normal, normal1, normal2, direction12, direction21
+    cdef Vector3 scaledDirection1, scaledDirection2, nearestPointA, nearestPointB
+    cdef double factor1, factor2
+    cdef Py_ssize_t i
+    for i in range(amount):
+        subVec3(&direction1, firstLineEnd.get(i), firstLineStart.get(i))
+        subVec3(&direction2, secondLineEnd.get(i), secondLineStart.get(i))
+        crossVec3(&normal, &direction1, &direction2)
+        if lengthVec3(&normal) > 1e-6:
+            valid.data[i] = True
+            crossVec3(&normal1, &direction1, &normal)
+            crossVec3(&normal2, &direction2, &normal)
+            subVec3(&direction12, firstLineStart.get(i), secondLineStart.get(i))
+            subVec3(&direction21, secondLineStart.get(i), firstLineStart.get(i))
+            factor1 = dotVec3(&direction21, &normal2)/ dotVec3(&direction1, &normal2)
+            factor2 = dotVec3(&direction12, &normal1)/ dotVec3(&direction2, &normal1)
+            firstParameter.data[i] = factor1
+            secondParameter.data[i] = factor2
+            scaleVec3(&scaledDirection1, &direction1, factor1)
+            scaleVec3(&scaledDirection2, &direction2, factor2)
+            addVec3(&nearestPointA, firstLineStart.get(i), &scaledDirection1)
+            addVec3(&nearestPointB, secondLineStart.get(i), &scaledDirection2)
+            nearestPoint1.data[i] = nearestPointA
+            nearestPoint2.data[i] = nearestPointB
+        else:
+            valid.data[i] = False
+            firstParameter.data[i] = 0
+            secondParameter.data[i] = 0
+            nearestPoint1.data[i] = Vector3(0,0,0)
+            nearestPoint2.data[i] = Vector3(0,0,0)
+    return nearestPoint1, nearestPoint2, firstParameter, secondParameter, valid
+
+def IntersectLineSphere(Py_ssize_t amount,
+                        VirtualVector3DList LineStart,
```

Comment:
please don't start variable names with a capital letter. I see this in multiple places here, should be corrected.